src/components/Datepicker/Datepicker.tsx

```diff
@@ -211,6 +213,13 @@ export const Datepicker: FC<DatepickerProps> = ({
     };
   }, [inputRef, datepickerRef, setIsOpen]);
 
+  useEffect(() => {
+    if (dateValue) {
+      setSelectedDate(dateValue);
+      setViewDate(dateValue);
+    }
+  }, [dateValue, setSelectedDate, setViewDate]);
```

Comment:
In addition to this, I think we have to initialise the state correctly on-mount to avoid flicker until this `useEffect` fires,  meaning we have to update:

```tsx
  // selectedDate is the date selected by the user
  const [selectedDate, setSelectedDate] = useState<Date>(defaultDate);
  // viewDate is only for navigation
  const [viewDate, setViewDate] = useState<Date>(defaultDate);
```

to

```tsx
  // selectedDate is the date selected by the user
  const [selectedDate, setSelectedDate] = useState<Date>(dateValue ?? defaultDate);
  // viewDate is only for navigation
  const [viewDate, setViewDate] = useState<Date>(dateValue ?? defaultDate);
```

Reply:
Yeah, I believe you are correct about that.

Reply:
Yeah, I agree. BTW, sorry to bring this discussion here, but I'm wondering how it would impact the "clear" button. Should it return to the default date? Should it set the date to `null`? Currently, we do have a bug/miss behavior for it... maybe we can fix it at this PR as well.

Reply:
@rluders `null` or `undefined` should do it, most date libraries do it this way

Reply:
Great guys, thanks for the review.